### PR TITLE
build: Migrate from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by yangchoo**
Migrates our existing CI/CD pipeline from Travis to GitHub actions.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    - name: Install Tox
+      run:  |
+        python -m pip install --upgrade pip
+        pip install tox
+
+    - name: Generate Docs
+      run: tox -e docs
+
+  build-and-publish:
```
Comment (yangchoo) [inline .github/workflows/ci.yml:65]: Build and publish step cribbed from various other sources, verified to be working in a separate repository.

**Comment by dzleidig** [comment]
Looks like..it works?
image

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,163 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - 'v[0-9]+.[0-9]+.[0-9]+'
```
Comment (dzleidig) [inline .github/workflows/ci.yml:8]: Nice, on tags too.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        MESSAGE="${MESSAGE//'%'/'%25'}"
+        MESSAGE="${MESSAGE//$'\n'/'%0A'}"
+        MESSAGE="${MESSAGE//$'\r'/'%0D'}"
+        echo ::set-output name=BODY::$MESSAGE
+
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (dzleidig) [inline .github/workflows/ci.yml:130]: Need to understand how these bindings work, where these vars are stored. My own research...

**Review by dzleidig — APPROVED** [review]
Approving if we need this ASAP, otherwise I (and others) can review in detail tomorrow.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (yangchoo) [reply .github/workflows/ci.yml:130]: Good general guide: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
Secrets: https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets

**Comment by yangchoo** [comment]
Thanks! No worries, it can certainly wait for a day. I'll wait till tomorrow then

**Pull request merged**